apis/Google.Cloud.Metadata.V1/Google.Cloud.Metadata.V1.IntegrationTests/Emulator/metadata_server_tests.py

```diff
@@ -0,0 +1,308 @@
+
+try:
+  server_address = 'http://' + os.environ[common.host_environment_variable]
+  request = urllib2.Request(server_address,
+                            headers={common.metadata_flavor : common.metadata_flavor_google})
+  urllib2.urlopen(request)
+except KeyError:
+  raise Exception('METADATA_EMULATOR_HOST must be set')
+except urllib2.URLError:
+  raise Exception('The emulator cannot be reached at %s' % server_address)
+
+error_count = 0
+
+def check_header(headers, url, key, value):
+  key_lower = key.lower()
+  if not key_lower in headers:
+    print "* Test failed: Cannot find %s header for %s" % (key, url)
+    return False
+
+  if headers[key_lower] != value:
+    print "* Test failed: Wrong %s header value for %s\n\tExpected: %s\n\tActual  : %s" % (key, url, value, headers[key_lower])
+    return False
+
+  return True
+
+def expect_content_absolute(url, expected, expected_content_type, expect_metadata_header=True):
+  global error_count
+  request = urllib2.Request(url, headers={common.metadata_flavor : common.metadata_flavor_google})
+  response = urllib2.urlopen(request)
+  headers = response.info().dict
+
+  failed = False
+  if expect_metadata_header:
+    failed |= not check_header(headers, url, common.metadata_flavor, common.metadata_flavor_google)
+  elif common.metadata_flavor.lower() in headers:
+    print "* Test failed: Was not expecting to find %s header for %s" % (common.metadata_flavor, url)
+    failed = True
+
+  failed |= not check_header(headers, url, common.server, common.server_value)
+  failed |= not check_header(headers, url, common.content_type, expected_content_type)
+  
+  contents = response.read()
+  if expected != contents:
+    failed = True
+    print "* Test failed: Wrong content for %s\n\tExpected: `%s`\n\tActual  : `%s`" % (url, expected, contents)
+    print "\tExpected type: %s\n\tActual type:   %s" % (type(expected), type(contents))
+
+  if not failed:
+    print "Test passed: %s" % url
+  else:
+    error_count += 1
+
+def expect_content(path, expected, expected_content_type, expect_metadata_header=True):
+  expect_content_absolute(server_address + '/computeMetadata/v1/' + path,
+                          expected,
+                          expected_content_type,
+                          expect_metadata_header)
+
+def expect_error(path, expected_code, metadata_flavor_value=common.metadata_flavor_google):
+  global error_count
+  url = server_address + '/computeMetadata/v1/' + path
+  request = urllib2.Request(url,
+                            headers={common.metadata_flavor :
+                                     metadata_flavor_value} if metadata_flavor_value else {})
+  failed = False
+  try:
+    response = urllib2.urlopen(request)
+    failed = True
+    print "* Test failed: Expected error for %s\n\tContent: %s" % (path, response.read())
+  except urllib2.HTTPError as e:
+    failed |= not check_header(e.headers, url, common.metadata_flavor, common.metadata_flavor_google)
```

Comment:
Yeah it really started as just some quick code I copied back and forth to a Compute Engine VM to ensure my assumptions about how the metadata server acted were correct, and that I was recreating it faithfully in the emulator. Then I realized it made sense to just clean it up and use it as integration tests for the emulator. Like I said, I'm pretty new to python and I didn't take the time to look into a good testing framework because it wasn't much code. But if you think there's a framework which would be ideal for this, I'd gladly change over to it in a subsequent PR.